replit_river/client_transport.py

```diff
@@ -286,32 +291,35 @@ async def _establish_handshake(
             )
         except FailedSendingMessageException:
             raise RiverException(
-                ERROR_CODE_STREAM_CLOSED, "Stream closed before response"
+                ERROR_CODE_STREAM_CLOSED,
+                "Stream closed before response, closing connection",
             )
+
         logging.debug("river client waiting for handshake response")
+        startup_grace_sec = 60
         try:
+            # listen even before we send
             response_msg = await asyncio.wait_for(
-                self._get_handshake_response_msg(websocket),
-                timeout=self._transport_options.session_disconnect_grace_ms / 1000,
+                self._get_handshake_response_msg(websocket), startup_grace_sec
```

Comment:
not strictly session disconnect grace, we also need to account for container startup time